tests/runtime/engine/runner.py

```diff
@@ -296,14 +302,23 @@ def run_test(test):
         finally:
             if befores:
                 for before in befores:
+                    try:
+                        befores_output.append(before.communicate(timeout=5))
```

Comment:
By here we already intend on killing the process. Does it make sense to wait up to 5s if it's still running? Maybe 1s or less is more prudent.